ui/Tabs.tsx

```diff
@@ -0,0 +1,61 @@
+"use client";
+
+import { Children, Fragment, ReactNode } from "react";
+import { Tab } from "@headlessui/react";
+import { ChipButton } from "@/ui";
+
+interface TabsProps {
+  children: ReactNode;
+  [key: string]: any;
+}
+
+export const TabGroup = ({ children, ...props }: TabsProps) => {
+  if (!children) return null;
+
+  const headerChildren = Children.toArray(children).filter(
+    child => (child as any)?.type === TabHeader
+  );
+  const bodyChildren = Children.toArray(children).filter(
+    child => (child as any)?.type === TabBody
+  );
+
+  return (
+    <Tab.Group {...props}>
+      {headerChildren}
+      {bodyChildren}
+    </Tab.Group>
+  );
+};
+
+export const TabHeader = ({
+  children,
+  ...props
+}: React.HTMLAttributes<HTMLDivElement>) => {
+  return (
+    <Tab.List className="flex space-x-3" {...props}>
+      {children}
+    </Tab.List>
+  );
+};
+
+export const TabBody = ({
+  children,
+  ...props
+}: React.HTMLAttributes<HTMLDivElement>) => {
+  return <Tab.Panels {...props}>{children}</Tab.Panels>;
+};
+
+export const TabStyled = ({ children, selected, ...props }: TabsProps) => {
```

Comment:
Here we can create a new interface which extends React.HTMLAttributes<HTMLDivElement> and adds selected prop.

```
interface TabStyledProps extends React.HTMLAttributes<HTMLDivElement> {
  selected?: boolean | null | undefined;
}
```

something like this works fine